pc_connect_warehouse_yellowcube/stock_picking_ext.py

```diff
@@ -117,54 +117,43 @@ def get_attachment_wab(self, cr, uid, ids, extension='pdf', context=None):
-                                'on picking with ID={1}, while just one was expected').format(len(att_picking_ids),
-                                                                                              stock_picking_out.id))
+        att_picking_ids = ir_attachment_obj.search(cr, uid, [
+            ('res_id', '=', stock_picking_out.id),
+            ('res_model', '=', 'stock.picking'),
+        ], order='create_date DESC', limit=number_of_attachments_to_use,
+                                                   context=context)
+
+        if len(att_picking_ids) == 0:
+            logger.warning(_('A bad number of picking reports was found '
+                             '({0}) on picking with ID={1}, '
+                             'while at least one was expected')
+                           .format(len(att_picking_ids), stock_picking_out.id))
             return result
 
-        # Determines if we have to attach also the picking report.
-        att_barcode_ids = []
-        if ir_module_obj.search(cr, uid, [('name', '=', 'pc_delivery_carrier_label_postlogistics'),
-                                          ('state', '=', 'installed'),
-                                          ], limit=1, count=True, context=context) \
-           and (not stock_picking_out.uses_bulkfreight):
-            att_barcode_ids = ir_attachment_obj.search(cr, uid, [('id', 'in', att_ids),
-                                                                 ('document_type', '=', 'barcode_out_report'),
-                                                                 ], context=context)
-            if len(att_barcode_ids) > 1:
-                raise Warning(_('A bad number of barcode reports was found ({0}) '
-                                'for picking with ID={1}, while one or none was expected').format(len(att_picking_ids),
-                                                                                                  stock_picking_out.id))
-
         # We compute the name of the output filename to be indicated on the WAB.
         output_filename = stock_picking_out.get_filename_for_wab(extension)
 
-        # Determines the ID of the attachment to send with the WAB. It will be only the ID of the picking
-        # if no barcode needs to be attached, BUT it will be *a new* attachment, created ad-hoc, with is the
-        # concatenation of the picking and the barcode.
-        if not att_barcode_ids:
-            attachment_to_send_id = att_picking_ids[0]  # Just the (only) delivery slip found.
+        if len(att_picking_ids) == 1:
+            attachment_to_send_id = att_picking_ids[0]
         else:
-            # Here we need to create a new attachment which is the concatenation of the picking and the barcode.
-            attachment_to_send_id = stock_picking_out._get_attachment_id_for_picking_and_barcode_concatenated(att_picking_ids[0], att_barcode_ids[0], extension)
+            attachment_to_send_id = stock_picking_out.\
```

Comment:
It makes easier to read on github, as very long lines mess with the web UI. Also, it helps when debugging on server